### ISSUE TITLE
Unmarshal error when copying object

### ISSUE THREAD

**Issue by jtopjian** [gophercloud/gophercloud#50]
I'm seeing the following error when trying to extract the results of a copy:

```
json: cannot unmarshal string into Go value of type int64
```

Here's the code I'm using:

```go
copyResult := objects.Copy(client, containerName, objectName, copyOpts)
copyHeader, err := copyResult.Extract()
```

Here's what `copyResult` looks like:

```
objects.CopyResult{HeaderResult:gophercloud.HeaderResult{Result:gophercloud.Result{Body:interface {}(nil), Header:http.Header{"Last-Modified":[]string{"Sun, 14 Aug 2016 23:11:50 GMT"}, "Etag":[]string{"4cb9bc8724551e6c7415f1080df84869"}, "X-Copied-From-Account":[]string{"AUTH_a54442e26ad8409a905659a268197eae"}, "Content-Type":[]string{"text/html; charset=UTF-8"}, "X-Trans-Id":[]string{"tx76234fbd166a462ca6e84-0057b0fab5"}, "Date":[]string{"Sun, 14 Aug 2016 23:11:49 GMT"}, "Content-Length":[]string{"0"}, "X-Copied-From":[]string{"test-container-Jy8w56q9/test-object-chsLSwuc"}, "X-Copied-From-Last-Modified":[]string{"Sun, 14 Aug 2016 23:11:50 GMT"}}, Err:error(nil)}}}
```

[ContentLength](https://github.com/gophercloud/gophercloud/blob/6dea3f17463923a8ea359d85ce2acaeb3ebb1c23/openstack/objectstorage/v1/objects/results.go#L262) is triggering the error

This is probably the same as the other unmarshal errors, but I'm not sure what to do since `CopyResult` is of type `gophercloud.HeaderResult`. Doing it incorrectly is probably going to break more things down the chain.

As well, `ContentLength` is defined as a string for the other structs. So maybe changing `int64` to `string` is the solution?

**Comment by jrperritt** [comment]
Same logic as [this](https://github.com/gophercloud/gophercloud/blob/master/openstack/compute/v2/servers/results.go#L169) should work, no?

**Comment by jrperritt** [comment]
I'd prefer to have types as they should be e.g. `ContentLength` as a number, times as `time.Time`s, ...

**Comment by jtopjian** [comment]
> Same logic as this should work, no?

Ah, yes. I see how this works now. I got side tracked into `HeaderResult`.

> I'd prefer to have types as they should be e.g. ContentLength as a number, times as time.Times, ...

Agreed. 

So as far as implementing this, I can take a look, but it might take a week or three for me to come back to. I can definitely do a quick PR of either changing `int64` to `string` (only to later be changed back) or implement Unmarshal on the CopyResult to work around this specific issue. Or just leave everything for a larger fix.

Thoughts?

**Comment by jrperritt** [comment]
I can fix this.

> Or just leave everything for a larger fix.

What would the larger fix be?

**Comment by jtopjian** [comment]
> What would the larger fix be?

Reviewing the types of all ObjectStorage headers that are currently strings and changing them to what their core type would be.

**Comment by jrperritt** [comment]
Ah. Yeah, that should probably be done too. I'll take care of it.

**Comment by jtopjian** [comment]
Sounds good! 😄

**Comment by jrperritt** [comment]
closed by #74
